src/state.ts

```diff
@@ -0,0 +1,24 @@
+import {Event} from './event';
+
+export class State {
+	public readonly events;
+
+	constructor() {
+		this.events = {};
+	}
+
+	async save(event: Event): Promise<unknown> {
+		if (event.key && event.value) {
+			await this.events.push(event);
```

Comment:
i didn’t see a push function in events. was events meant to be an array of type Event?